lib/std/start.zig

```diff
@@ -603,14 +603,29 @@ pub fn callMain() u8 {
 }
 
 pub fn call_wWinMain() std.os.windows.INT {
+    const peb = std.os.windows.peb();
     const MAIN_HINSTANCE = @typeInfo(@TypeOf(root.wWinMain)).Fn.params[0].type.?;
     const hInstance = @as(MAIN_HINSTANCE, @ptrCast(std.os.windows.kernel32.GetModuleHandleW(null).?));
     const lpCmdLine = std.os.windows.kernel32.GetCommandLineW();
 
-    // There's no (documented) way to get the nCmdShow parameter, so we're
-    // using this fairly standard default.
-    const nCmdShow = 5;
+    // This makes Zig mostly match the nCmdShow behavior of a C program with a WinMain symbol:
+    // - With STARTF_USESHOWWINDOW set in STARTUPINFO.dwFlags of the CreateProcess call:
+    //   - Compiled with subsystem:console -> nCmdShow is always SW_SHOWDEFAULT
+    //   - Compiled with subsystem:windows -> nCmdShow is STARTUPINFO.wShowWindow from
+    //     the parent CreateProcess call
+    // - With STARTF_USESHOWWINDOW unset:
+    //   - nCmdShow is always SW_SHOWDEFAULT
+    //
+    // TODO: Figure out how to check the subsystem and use it to match the above
+    //       behavior, or intentionally avoid matching the above behavior with
+    //       regards to the subsystem.
```

Comment:
`std.builtin.subsystem`

Reply:
Of course! I was way overthinking things.

Reply:
After trying this, `std.builtin.subsystem` does not actually match up with the subsystem put into the `.exe`, as the linker that writes the OptionalHeader only cares about the `--subsystem` CLI flag, and the `builtin.explicit_subsystem` field that `std.builtin.subsystem` looks for is not set by anything AFAICT.

I've created a new issue for this: https://github.com/ziglang/zig/issues/17808